### PR TITLE
adds personalities

### PR DESCRIPTION
**Pull request by michaelhillebrand**

### CODE REVIEW

File: discord_bot.py [hunk 1]
```diff
@@ -1,9 +1,30 @@
...
+        'hank': Hank,
+        'bojack': BoJack
+    }
+
+    last_personality_change = datetime.datetime.utcnow()
 
-    async def on_ready(self):
+    async def on_ready(self) -> None:
+        try:
+            await self.user.edit(avatar=None)
```
Comment (willpoorman) [inline discord_bot.py]: Any reason wer're not setting the avatar here like we do in the change_personality command? [cogs/personality.py#L38](https://github.com/michaelhillebrand/huskiebot/pull/23/files#diff-caba36e032f2a9f4f77022de0f23951bR38)

File: personalities/hank_hill.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        "You know whats not cool Bobby? Hell.",
+        "Do I Look Like I Know What a JPEG is?",
+        "I'm so tired of you South Texas Pig Jockeys comin' in here, wakin' me up to say \"How 'bout them Cowboys!\". "
+        "Arlen stinks and Wichita Falls rules! And you know why? Cause in five minutes I can be in the Great State of "
+        "Oklahoma! Go Sooners!",
+        "This tornado's already registered a level two on the Fujisaki scale. A storm that strong will send an egg "
+        "through a barn door - two barn doors if one of them's open",
+        "I know what's wrong with it. It's a Ford. You know what they say Ford stands for, don't ya? It stands for "
+        "'Fix it again, Tony'",
+    )
```
Comment (willpoorman) [inline personalities/hank_hill.py]: No newline at end of file

File: cogs/presence_changer.py [hunk 3]
```diff
@@ -47,11 +35,11 @@ def cog_unload(self):
             discord.Activity
         """
         try:
-            logging.info(f"changing presence to: type: {self.STATUSES[self.index][0]}, "
-                         f"name: {self.STATUSES[self.index][1]}")
+            logging.info("changing presence to: type: {type}, name: {name}".format(type=self.bot.current_personality.presence_options[self.index][0],
```
Comment (michaelhillebrand) [inline cogs/presence_changer.py:38]: set self.bot.current_personality.presence_options to variable

File: discord_bot.py (hunk 1 above)
Comment (michaelhillebrand) [reply discord_bot.py:25]: I will need to write something to have the personality to persist

File: personalities/__init__.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+from os.path import abspath, join, dirname
+
+BASE_PATH = dirname(dirname(abspath(__file__)))
```
Comment (michaelhillebrand) [inline personalities/__init__.py:3]: This is duplicating variables.

File: run.py [hunk 5]
```diff
@@ -40,7 +40,7 @@ def setup_bot():
         DankMemes,
         PresenceChanger,
         ChatModerator,
-        GruNosePoster
```
[the remark this answers is not in the document]
Comment (michaelhillebrand) [reply run.py:43]: This should be abstracted to some "auto poster"

File: utils/settings.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        key: str
+
+        Returns
+        -------
+        None
+        """
+        logging.debug(f'Getting setting for {key}')
+        return self.data[key]  # Intentional error is thrown
+
+    def save(self) -> None:
```
Comment (willpoorman) [inline utils/settings.py:54]: Another instance where we're saying it doesn't return anything, but yet it's returning something

File: utils/settings.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        Parameters
+        ----------
+        key: str
+        value: any
+
+        Returns
+        -------
+        None
+        """
+        logging.debug('Updating settings')
```
Comment (willpoorman) [inline utils/settings.py:71]: Maybe change to
```suggestion
        logging.debug(f'Updating setting: {key}')
```
or
```suggestion
        logging.debug(f'Updating setting: {key} to {value}')
```

File: utils/settings.py [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+        try:
+            self.data[key] = value
+            self._save()
+        except Exception as e:
+            # catch values that can't be pickled
+            self.data = old_setting
+            raise e
+
+    def update(self, data: dict) -> None:
+        """Updates settings with given data.
```
Comment (willpoorman) [inline utils/settings.py:87]: Add more info explaining what this does and how it differs from `set`. Explain how it merges the data param with the classes data dict to update only the values from the data param

File: utils/settings.py [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+    def get(self, key: str):
+        """Gets current value for key in settings.
+
+        Parameters
+        ----------
+        key: str
+
+        Returns
+        -------
+        None
```
[the remark this answers is not in the document]
Comment (michaelhillebrand) [reply utils/settings.py:49]: Yes, we can look into this. This was a typo in the docstring

File: utils/settings.py (hunk 6 above)
Comment (michaelhillebrand) [reply utils/settings.py:54]: that is a typo, self._save() doesn't return anything

File: run.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (willpoorman) [reply run.py]: Created Issue #38 to track this abstraction/conversion

File: discord_bot.py (hunk 1 above)
Comment (willpoorman) [reply discord_bot.py]: Resolved by adding a settings class in commit 7577f23

File: utils/settings.py (hunk 6 above)
Comment (willpoorman) [reply utils/settings.py:59]: Resolved in 4c5b1bf

File: utils/settings.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (willpoorman) [reply utils/settings.py]: Resolved in 4c5b1bf

File: utils/settings.py (hunk 8 above)
Comment (willpoorman) [reply utils/settings.py]: Resolved in 4c5b1bf

**Pull request merged**